modules/core/core-schema/src/main/java/com/enonic/xp/core/impl/schema/mapper/FormOptionSetOptionsDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.enonic.xp.core.impl.schema.mapper;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.enonic.xp.form.FormOptionSetOption;
+
+public final class FormOptionSetOptionsDeserializer
+    extends JsonDeserializer<Iterable<FormOptionSetOption>>
+{
+    @Override
+    public Iterable<FormOptionSetOption> deserialize( final JsonParser jsonParser, final DeserializationContext deserializationContext )
+        throws IOException
+    {
+        final ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
+        final ArrayNode arr = mapper.readTree( jsonParser );
+
+        for ( JsonNode node : arr )
+        {
+            if ( node.isObject() && !node.has( "type" ) )
```

Comment:
formoptionset allows only formoptionsetption
this code assumes that it is possible to add another type if specified

Reply:
It necessary for `FormItemMapper` which marked with
`@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", visible = true)` . But in yml we do not need to use `type` for `OptionSetOption`. Check tests please